lib/doctordata.rb

```diff
@@ -1,22 +1,52 @@
 require "doctordata/version"
 require 'csv'
 require 'roo'
-require 'faraday'
 
 module Doctordata
   class Parser
     class << self
       def from_csv_table(table)
         # there is much room to do performance tuning
-        table.
-          map do |row|
-            row.
-              reject { |k, v| v == nil || v == '' }.
-              reject { |k, v| k == nil || k == '' || k.start_with?('#') }.
-              map { |k, v| "#{k}=#{v}" }.
-              join('&')
-          end.
-          map { |s| Faraday::NestedParamsEncoder.decode(s) }
+        checked_table = table.by_col!.delete_if{ |k, v| k == nil || k == '' || k.start_with?('#') }
```

Comment:
argument object should not be destructive, but i'll think it later so it's okay this time.